awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/OtelJMXMetricsCustomizerProvider.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.javaagent.providers;
+
+import io.opentelemetry.exporter.otlp.http.metrics.OtlpHttpMetricExporter;
+import io.opentelemetry.exporter.otlp.internal.OtlpConfigUtil;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.DefaultConfigProperties;
+import io.opentelemetry.sdk.metrics.Aggregation;
+import io.opentelemetry.sdk.metrics.InstrumentSelector;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.metrics.View;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * You can control when these customizations are applied using the property otel.jmx.enabled or the
+ * environment variable OTEL_JMX_ENABLED_CONFIG. This flag is disabled by default.
+ */
+public class OtelJMXMetricsCustomizerProvider implements AutoConfigurationCustomizerProvider {
+    private static final Duration DEFAULT_METRIC_EXPORT_INTERVAL = Duration.ofMinutes(1);
+    private static final Logger logger =
+            Logger.getLogger(OtelJMXMetricsCustomizerProvider.class.getName());
+
+    private static final String OTEL_JMX_ENABLED_CONFIG = "otel.jmx.enabled";
+    private static final String OTEL_JMX_ENDPOINT_CONFIG = "otel.jmx.exporter.metrics.endpoint";
```

Comment:
I'm still of the opinion that both the `otel.jmx.enabled` property and the endpoint property should be set to enable the customizer. Enabling `otel.jmx.enabled` does not necessarily mean that the exporter should be set up. Customers could use the default exporter.

Reply:
updated to check with both ```otel.jmx.enabled``` and ```otel.jmx.exporter.metrics.endpoint```